hazelcast/sql.py

```diff
@@ -1167,32 +1203,18 @@ def __init__(self, connection_manager, serialization_service, invocation_service
         self._serialization_service = serialization_service
         self._invocation_service = invocation_service
 
-    def execute(self, sql, *params):
+    def execute(self, sql, params, kwargs):
         """Constructs a statement and executes it.
 
         Args:
             sql (str): SQL string.
-            *params: Query parameters.
-
-        Returns:
-            hazelcast.future.Future[SqlResult]: The execution result.
-        """
-        statement = SqlStatement(sql)
-
-        for param in params:
-            statement.add_parameter(param)
-
-        return self.execute_statement(statement)
-
-    def execute_statement(self, statement):
-        """Executes the given statement.
-
-        Args:
-            statement (SqlStatement): The statement to execute.
+            params (tuple): Query parameters.
+            kwargs (dict): Arguments to customize the query.
 
         Returns:
             hazelcast.future.Future[SqlResult]: The execution result.
         """
+        statement = _SqlStatement(sql, params, **kwargs)
```

Comment:
For the reviewers, I construct the statement outside of the try-except block below so that, for validation errors, we immediately throw the exception not ImmediateFutureException